src/include/mutex.h

```diff
@@ -99,7 +99,7 @@ struct __wt_rwlock { /* Read/write lock */
 struct __wt_spinlock {
 #if SPINLOCK_TYPE == SPINLOCK_GCC
     WT_CACHE_LINE_PAD_BEGIN
-    volatile int lock;
+    volatile bool lock;
```

Comment:
From the GCC documentation:

> It should be only used for operands of type bool or char. For other types only part of the value may be set.

According to [this](https://github.com/wiredtiger/wiredtiger/pull/4740#discussion_r305622159) comment from @keithbostic , we probably don't need to worry about this in WiredTiger. But I don't believe there's any reason not to use `bool` so let's just follow their advice.

Reply:
I'm neutral on this. It probably does not change anything because this is the beginning of the structure and we're using `WT_CACHE_LINE_PAD_BEGIN` to start it aligned. Then the next item after `lock` is a pointer, `const char *name` which will be 8 bytes and aligned anyway. So we still have that 7 bytes. If we care about bytes here or we think we'd ever want any other debug or info, you could add in a `uint8_t unused[7];` to make things very clear.

Reply:
91f4d23

Yeah agreed. It almost certainly isn't a problem as the functions are likely reading and writing the first byte of the `int`. But I figure we may as well follow their advice since it explicitly says it should only be used for `bool` or `char`.